beacon_node/http_api/src/block_id.rs

```diff
@@ -252,23 +255,30 @@ impl BlockId {
         }
     }
 
-    /// Return the `BlobSidecarList` identified by `self`.
-    pub fn blob_sidecar_list<T: BeaconChainTypes>(
+    #[allow(clippy::type_complexity)]
+    pub fn get_blinded_block_and_blob_list_filtered<T: BeaconChainTypes>(
         &self,
+        indices: BlobIndicesQuery,
         chain: &BeaconChain<T>,
-    ) -> Result<BlobSidecarList<T::EthSpec>, warp::Rejection> {
-        let root = self.root(chain)?.0;
-        chain
+    ) -> Result<
+        (
+            SignedBlindedBeaconBlock<T::EthSpec>,
+            BlobSidecarList<T::EthSpec>,
+            ExecutionOptimistic,
+            Finalized,
+        ),
+        warp::Rejection,
+    > {
+        let (root, execution_optimistic, finalized) = self.root(chain)?;
+        let block = BlockId::blinded_block_by_root(&root, chain)?.ok_or_else(|| {
+            warp_utils::reject::custom_not_found(format!("beacon block with root {}", root))
+        })?;
+
+        // Return the `BlobSidecarList` identified by `self`.
+        let blob_sidecar_list = chain
```

Comment:
Now that we're loading the block and blobs together, it could be good to check that all the blobs are loaded, and return a 404 in case they aren't

Related to this issue:

- https://github.com/sigp/lighthouse/issues/5640

I added the `backwards-incompat` tag because I thought this chance may already accomplish this, but we don't have the check (yet), so it doesn't.